### PR TITLE
#11 Feature: implement unique user emails

### LINKED ISSUE #11: Feature: Implement unique user emails
For Parent and Child
Validation during:
- [x] add User (checking if it exists)

- [x] delete User (removal from User and Parent tables; removal from User and Child tables) 

- [x] update User (checking if it exists)

### PR DESCRIPTION
**Pull request by AntoninaMJ**
Added new IllegalArgumentExceptions if there email already exists in database.
During: addParent, addChild, updateParent, updateChild.
Implemented user removing before deleteParent and deleteChild (Transactional).
Added test for these features.

### CODE REVIEW

File: src/main/java/pl/antonina/tasks/child/ChildService.java [hunk 1]
```diff
@@ -34,11 +38,16 @@ ChildView getChild(long id) {
     }
 
     void addChild(long parentId, ChildData childData) {
-        Parent parent = parentRepository.findById(parentId).orElseThrow();
+        String email = childData.getUserData().getEmail();
+        boolean userExists = userRepository.findByEmail(email).isPresent();
+        if (userExists) {
+            throw new IllegalArgumentException("User with given email already exists");
```
Comment (malpiszon) [inline src/main/java/pl/antonina/tasks/child/ChildService.java]: Please consider:

using own type of exception that will be mapped later to proper exception or directly the error code (https://www.baeldung.com/exception-handling-for-rest-with-spring)
adding throws IllegalArgumentException (or your own Exception name) to the method signature - it's part of this method flow, something that is quite expected. :)

**Review by malpiszon — COMMENTED** [review]
Please take a look, maybe it's worth to improve. :)

File: src/main/java/pl/antonina/tasks/child/ChildService.java (hunk 1 above)
Comment (MartaSlysz) [reply src/main/java/pl/antonina/tasks/child/ChildService.java]: I can agree :)

Comment (AntoninaMJ) [reply src/main/java/pl/antonina/tasks/child/ChildService.java]: Done :)

**Pull request merged**